plugins/test/unit/plugins/importers/yum/parse/test_rpm.py

```diff
@@ -30,3 +36,23 @@ def test_non_supported_encoding(self):
         start_string.decode.side_effect = UnicodeError()
         result_string = rpm.string_to_unicode(start_string)
         self.assertEquals(None, result_string)
+
+
+class PackageHeaders(unittest.TestCase):
+    """
+    tests for package headers and signature extraction
+    """
+
+    def test_package_signature_from_header(self):
+        sample_rpm_filename = os.path.join(DATA_DIR, 'walrus-5.21-1.noarch.rpm')
+        headers = rpm.package_headers(sample_rpm_filename)
+        signature = rpm.package_signature(headers)
+        self.assertEquals(signature, 'f78fb195')
+        self.assertEquals(len(signature), 8)
+        self.assertTrue(isinstance(headers, rpm_module.hdr))
```

Comment:
Since this asserts that the test setup went correctly, I would make this assertion before doing the test (which is to call ``package_signature``. Assuming something went wrong when getting the ``headers`` value, that would likely cause the call to ``package_signature`` to fail, and thus the test would fail before ever getting to this assertion.